token/program-2022/src/extension/group_member_pointer/processor.rs

```diff
@@ -0,0 +1,145 @@
+use {
+    crate::{
+        check_program_account,
+        error::TokenError,
+        extension::{
+            group_member_pointer::{
+                instruction::{
+                    GroupMemberPointerInstruction, InitializeInstructionData, UpdateInstructionData,
+                },
+                GroupMemberPointer,
+            },
+            BaseStateWithExtensions, StateWithExtensions, StateWithExtensionsMut,
+        },
+        instruction::{decode_instruction_data, decode_instruction_type},
+        processor::Processor,
+        state::Mint,
+    },
+    solana_program::{
+        account_info::{next_account_info, AccountInfo},
+        entrypoint::ProgramResult,
+        msg,
+        program_error::ProgramError,
+        pubkey::Pubkey,
+    },
+    spl_pod::optional_keys::OptionalNonZeroPubkey,
+    spl_token_group_interface::state::TokenGroup,
+};
+
+fn check_group_update_authority(
+    group_info: &AccountInfo,
+    group_update_authority_info: &AccountInfo,
+) -> Result<(), ProgramError> {
+    if !group_update_authority_info.is_signer {
+        msg!("Group update authority must be a signer");
+        Err(ProgramError::MissingRequiredSignature)?;
+    }
+    let group_data = group_info.data.borrow();
+    let group_mint = StateWithExtensions::<Mint>::unpack(&group_data)?;
+    let group_state = group_mint.get_extension::<TokenGroup>()?;
+    if Option::<Pubkey>::from(group_state.update_authority)
+        != Some(*group_update_authority_info.key)
+    {
+        msg!("Incorrect update authority for group");
+        Err(TokenError::OwnerMismatch)?;
+    }
+    Ok(())
+}
```

Comment:
This doesn't work with an external group program, where we can't figure out the authority, so we should probably get rid of the group authority requirement. Let's let any mint point at any member account, and it's up to clients to make sure that the accounts point at each other, same as with token metadata